qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +289,120 @@ def default_values(self) -> List[DefaultCalValue]:
                     defaults.append(DefaultCalValue(value, param.name, tuple(), name))
 
         return defaults
+
+
+class FixedFrequencyTransmonCR(FixedFrequencyTransmon):
```

Comment:
How big is the overhead of defining EcoedCR and CR as separate libraries? I think _library_ is a collection of static data, not something directly configured. Probably you can remove `echo` option and rename this class to `FixedFrequencyTransmonEchoedCR`. Likely `echo=False` will be never used because large ZI interaction cannot be cancelled without hard-coded Rz gate, which we need definition of the frame to implement. @wshanks any thoughts?

Reply:
I also prefer to separate CR library from the single qubit and instantiating the calibration with something like `[FixedFrequencyTransmonEchoedCR, FixedFrequencyTransmonSingleQubits]`. This inheritance will cause conflict of single qubit definition if we want to define mixture of ECR and DDCX in the same calibration set. Also the reusability of the library is limited, i.e. we may want to define another type of single qubit gate. The library should be defined for the minimum set. 

I guess current design is to overcome missing echo gate, but we need to implement some mechanism to externally feed dependent calibrations. For example, `_build_schedule` doesn't need to be called immediately. This means, `Calibrations` can call the build method with dictionary that defines externally defined subroutine.

```python
self _build_schedule(self, ..., external_dependency: Dict[str, ScheduleBlock]):
    if "some_gate" not in external_dependency:
        raise SomeError("I need some gate!")
    with pulse.build() as sched:
        pulse.call(external_dependency["some_gate"])
```

Reply:
See here: [5d6a195](https://github.com/Qiskit/qiskit-experiments/pull/725/commits/5d6a19574d6e67a3a179298bbeb339b136f8533c) the echo is now backed into the class. Also initializing the cals form a list of libraries is implemented. This allows us to us `Call` in a proper way across multiple libraries as you suggested.